pkg/core/region_tree_test.go

```diff
@@ -368,6 +368,45 @@ func TestRandomRegionDiscontinuous(t *testing.T) {
 	checkRandomRegion(re, tree, []*RegionInfo{regionA, regionB, regionC, regionD}, []keyutil.KeyRange{keyutil.NewKeyRange("", "")})
 }
 
+func TestStoreRegionCount(t *testing.T) {
+	re := require.New(t)
+	regions := NewRegionsInfo()
+	i := uint64(1)
+	voterFn := func() *metapb.Peer {
+		i++
+		return &metapb.Peer{
+			StoreId: 2,
+			Id:      i,
+			Role:    metapb.PeerRole_Voter,
+		}
+	}
+	learnerFn := func() *metapb.Peer {
+		i++
+		return &metapb.Peer{
+			StoreId: 3,
+			Id:      i,
+			Role:    metapb.PeerRole_Learner,
+		}
+	}
+
+	regions.CheckAndPutRegion(NewTestRegionInfo(1, 1, []byte("a"), []byte("c"), WithAddPeer(voterFn()), WithAddPeer(learnerFn())))
+	regions.CheckAndPutRegion(NewTestRegionInfo(2, 1, []byte("e"), []byte("g"), WithAddPeer(voterFn()), WithAddPeer(learnerFn())))
+	regions.CheckAndPutRegion(NewTestRegionInfo(3, 1, []byte("g"), []byte("i"), WithAddPeer(voterFn()), WithAddPeer(learnerFn())))
+	for _, endKey := range [][]byte{[]byte("b"), []byte("c"), []byte("d"), []byte("e"), []byte("f"), []byte("")} {
```

Comment:
Do we need some hole cases such as [c,e], [b,d]

Reply:
done